src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -818,4 +862,68 @@ private function isMethodAccessible(\ReflectionClass $class, $methodName, $param
 
         return false;
     }
+
+    /**
+     * Gets a PropertyPath instance and caches it.
+     *
+     * @param string|PropertyPath $propertyPath
+     *
+     * @return PropertyPath
+     */
+    private function getPropertyPath($propertyPath)
+    {
+        if ($propertyPath instanceof PropertyPathInterface) {
+            // Don't call the copy constructor has it is not needed here
+            return $propertyPath;
+        }
+
+        if (isset($this->propertyPathCache[$propertyPath])) {
+            return $this->propertyPathCache[$propertyPath];
+        }
+
+        if ($this->cacheItemPool) {
+            $item = $this->cacheItemPool->getItem(self::CACHE_PREFIX_PROPERTY_PATH.$propertyPath);
+            if ($item->isHit()) {
+                return $this->propertyPathCache[$propertyPath] = $item->get();
+            }
+        }
+
+        $propertyPathInstance = new PropertyPath($propertyPath);
+        if (isset($item)) {
+            $item->set($propertyPathInstance);
+            $this->cacheItemPool->save($item);
+        }
+
+        return $this->propertyPathCache[$propertyPath] = $propertyPathInstance;
+    }
+
+    /**
+     * Creates the APCu adapter if applicable.
+     *
+     * @param string               $namespace
+     * @param int                  $defaultLifetime
+     * @param string               $version
+     * @param LoggerInterface|null $logger
+     *
+     * @return AdapterInterface
+     *
+     * @throws RuntimeException When the Cache Component isn't available
+     */
+    public static function createCache($namespace, $defaultLifetime, $version, LoggerInterface $logger = null)
+    {
+        if (!class_exists('Symfony\Component\Cache\Adapter\ApcuAdapter')) {
+            throw new \RuntimeException(sprintf('The Cache Component must be installed to use the "%s::%s" factory.', __CLASS__, __METHOD__));
```

Comment:
`sprintf('The Symfony Cache component must be installed to use %s().', __METHOD__)`